### ISSUE TITLE
Functions must have an explicit list of parameters. Try adding a parameter list.

### ISSUE THREAD

**Issue by chentianxin** [flutterchina/dio#892]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | ios / flutter                         |      |
| Platform Version               |13.4.1 / 1.20.0                             |      |
| Dio Version                    | ^3.0.10                                 |      |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

```dart
BaseOptions options = BaseOptions(
  baseUrl: Global.dioBaseUrl,
);
Dio dio = Dio(options);
dio.interceptors.add(ErrorCodeInterceptors());
```
![image](https://user-images.githubusercontent.com/49486355/89854779-5b99c380-dbc7-11ea-907b-2493e4c66a50.png)

**Comment by ThinkDigitalSoftware** [comment]
Are you sure your analyzer is refreshed? The underline doesn't look like it's reading the right line. try reopening the project first, just in case this is an issue with it being out of sync

**Comment by chentianxin** [comment]
> 您确定分析仪已刷新吗？下划线看起来好像不是在读正确的线。尝试先重新打开项目，以防万一这是一个不同步的问题



> 您确定分析仪已刷新吗？下划线看起来好像不是在读正确的线。尝试先重新打开项目，以防万一这是一个不同步的问题

Restarting vscode is the same error

**Comment by ThinkDigitalSoftware** [comment]
What's the type of your class?

**Comment by chentianxin** [comment]
> What's the type of your class?

```dart
BaseOptions options = BaseOptions(
  baseUrl: Global.dioBaseUrl,
);
Dio dio = Dio(options);

// dio.interceptors.add(ErrorCodeInterceptors());
// dio.interceptors.add(new TokenInterceptor());
class ErrorCodeInterceptors extends InterceptorsWrapper {
  @override
  Future onRequest(RequestOptions options) {
    print("REQUEST[${options?.method}] => PATH: ${options?.path}");
    return super.onRequest(options);
  }

  @override
  Future onResponse(Response response) {
    print(
        "RESPONSE[${response?.statusCode}] => PATH: ${response?.request?.path}");
    return super.onResponse(response);
  }

  @override
  Future onError(DioError err) {
    print("ERROR[${err?.response?.statusCode}] => PATH: ${err?.request?.path}");
    return super.onError(err);
  }
}
```

**Comment by ThinkDigitalSoftware** [comment]
Don't know. Looks fine to me. I still think it's an analyzer error

**Comment by ruikc** [comment]
`
var dioOptions = BaseOptions(
  baseUrl: baseUrl,
  connectTimeout: connectTimeout,
  receiveTimeout: receiveTimeout,
);
var _dio = Dio();
_dio.interceptors.add(CustomInterceptors());
`
how to solute?
![image](https://user-images.githubusercontent.com/7445875/120458575-f8815380-c3c9-11eb-883b-96ca45f8486b.png)
![image](https://user-images.githubusercontent.com/7445875/120458658-0a62f680-c3ca-11eb-96b3-cdb007f9d850.png)

**Comment by wil-x** [comment]
原因是不能在 dart 模块中直接执行代码，模块内的逻辑应该包含在函数或类中。

**Comment by energynumbers** [comment]
translation of ghost's comment, for anyone else who needs it: 

The reason is that code cannot be executed directly in the dart module.

The logic within the module should be contained in functions or classes.

So, this fails: `_dio.interceptors.add(CustomInterceptors());`

But this works:
```
void init_my_dio() {
  _dio.interceptors.add(CustomInterceptors());
}
```
